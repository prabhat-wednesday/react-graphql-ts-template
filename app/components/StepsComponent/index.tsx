import React from 'react';
import styled from 'styled-components';
import { Steps } from 'antd';

const CustomStep = styled(Steps)`
  padding: 0 10rem;
`;
const { Step } = Steps;

interface StepsComponentProps {
  currentIndex: Number;
  stepsMsg: { title: string }[];
}
const StepsComponent = ({ currentIndex, stepsMsg }: StepsComponentProps) => {
  return (
    <CustomStep current={currentIndex}>
      {stepsMsg.map((item: any) => (
        <Step key={item.title} {...item} />
      ))}
    </CustomStep>
  );
};

export default StepsComponent;
