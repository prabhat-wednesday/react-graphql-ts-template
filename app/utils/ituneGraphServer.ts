import ApolloClient, { DocumentNode, InMemoryCache } from 'apollo-boost';

export const client = new ApolloClient({
  uri: 'https://itunes.apple.com/search?term=',
  cache: new InMemoryCache()
});

export interface GqlQueryReponse<Data> {
  data?: Data;
  error?: any;
  ok: boolean;
}

export const getQueryResponse = <Data, Variables>(
  query: DocumentNode,
  variables?: Variables
): Promise<GqlQueryReponse<Data>> => {
  console.log(query, 'in graphql server');
  return client
    .query<Data, Variables>({ query, variables })
    .then((res) => {
      console.log(res, 'in graphql');
      if (res.errors) {
        console.log(res);
        return { error: res.errors, ok: false };
      } else {
        console.log(res);
        return { data: res.data, ok: true };
      }
    })
    .catch((err) => {
      return { error: err, ok: false };
    });
};
