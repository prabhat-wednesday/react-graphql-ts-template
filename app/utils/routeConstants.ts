export type RouteConstant = {
  route: string;
  exact?: boolean;
  isProtected?: boolean;
  props?: object;
};

const routeConstants: Record<string, RouteConstant> = {
  itune: {
    route: '/',
    exact: true
  },
  song: {
    route: '/song/:trackId',
    exact: true
  }
};

export type RouteKeys = keyof typeof routeConstants;

export default routeConstants;
