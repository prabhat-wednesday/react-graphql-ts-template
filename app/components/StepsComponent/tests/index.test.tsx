import React from 'react';
import { renderProvider } from '@app/utils/testUtils';
import StepsComponent from '..';

describe('<StepsComponent/> ', () => {
  const defaultProps = {
    currentIndex: 1,
    stepsMsg: [{ title: 'Username' }]
  };
  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(<StepsComponent {...defaultProps} />);
    expect(baseElement).toMatchSnapshot();
  });
});
