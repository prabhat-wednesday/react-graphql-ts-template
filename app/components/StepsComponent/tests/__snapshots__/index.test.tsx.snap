// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepsComponent/>  should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="ant-steps ant-steps-horizontal StepsComponent__CustomStep-ymgfaa-0 kQDDmx ant-steps-label-horizontal"
    >
      <div
        class="ant-steps-item ant-steps-item-finish"
      >
        <div
          class="ant-steps-item-container"
        >
          <div
            class="ant-steps-item-tail"
          />
          <div
            class="ant-steps-item-icon"
          >
            <span
              class="ant-steps-icon"
            >
              <span
                aria-label="check"
                class="anticon anticon-check ant-steps-finish-icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="check"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 00-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <div
            class="ant-steps-item-content"
          >
            <div
              class="ant-steps-item-title"
            >
              Username
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
